src/main/java/io/jenkins/plugins/sprp/YAML_FlowDefinition.java

```diff
@@ -125,6 +138,36 @@ public FlowExecution create(FlowExecutionOwner owner, TaskListener listener,
         return new CpsFlowExecution(script, false, owner);
     }
 
+    private List<String> getCleanRefspecs(List<UserRemoteConfig> userRemoteConfigs){
+        List<String> refSpecs = new ArrayList<>();
+
+        for(UserRemoteConfig urc: userRemoteConfigs)
+            for(String s: urc.getRefspec().split("\\+", 0))
+                if(!s.equals(""))
+                    refSpecs.add(s);
+
+        for(int i = 0; i < refSpecs.size(); i++)
+            refSpecs.set(i, refSpecs.get(i).trim());
```

Comment:
I feel the second loop is not needed if the first loop trims `s` before it is added.

Also, do not use single letter variable names, it makes it unsearchable (there are many hits on the letter s in the file). You could perhaps write `for (String refspec : urc.getRefSpec().split("\\+", 0).trim()) { ... }`.

Lastly, I do not know if Jenkins has coding style guidelines, but I suggest using curly braces even when they are optional.

Reply:
In case of "+ someThing", the first loop will generate " someThing" hence there will be a space in the beginning. Hence second loop will be needed.

Reply:
I still think @martinda is right, the 2nd loop doesn't seem to be needed

currently your first for loop is like this:
```
String s: urc.getRefspec().split("\\+", 0)
```
and you say that `s = urc.getRefspec()` will give you "+ someThing"
but then you apply `s.split("\\+", 0)` to get " someThing"
and if you then apply `s.trim()` you'll get what you want with "someThing"

Reply:
Ok, I have tested this thing, and I agree that the second loop in not needed. Hence it has been removed.